docs/reference/generated/scroll-area-root.json

```diff
@@ -12,7 +18,26 @@
       "detailedType": "| ReactElement\n| ((\n    props: HTMLProps,\n    state: ScrollArea.Root.State,\n  ) => ReactElement)"
     }
   },
-  "dataAttributes": {},
+  "dataAttributes": {
+    "data-has-overflow-x": {
+      "description": "Present when the scroll area content is wider than the viewport."
+    },
+    "data-has-overflow-y": {
+      "description": "Present when the scroll area content is taller than the viewport."
+    },
+    "data-overflow-x-end": {
```

Comment:
What's the advantage of having separate attributes for -start and -end?
We could use `data-has-overflow-x="start end"`.

Reply:
Edge scroll gradients work better with boolean attrs.

Compare:

```css
.fade-start[data-overflow-y-start],
.fade-end[data-overflow-y-end] {
  opacity: 1;
}
```

This breaks unless you use `~=` token matchers which isn't intuitive:

```css
.fade-start[data-has-overflow-y="start"],
.fade-end[data-has-overflow-y="end"] {
  opacity: 1;
}
```